azure-pipelines.yml

```diff
@@ -60,6 +59,14 @@ jobs:
     - script: go test $(dirname $(find . -path ./vendor -prune -o -path ./sdk -prune -o -path ./sdk -prune -o -name '*_test.go' -print) | sort -u)
       workingDirectory: '$(sdkPath)'
       displayName: 'Run Tests'
+    - task: PowerShell@2
```

Comment:
Why aren't we using the template here?

Reply:
@jhendrixMSFT is the the best place to add in a link verification check?

Reply:
1. The yml here reads the $sdkPath to get the script. We are using ${Build.SourceDirectory} in template. The two paths are not the same 
2. We have to exclude path of 'vendor' when read .md files. So the one in template cannot apply this case.

Reply:
I'm not very familar with the go repo it doesn't even look like this pipeline is for the track 2 libraries which I believe are under the root sdk folder. So it then becomes a question of should we run the link validation on these libraries.

If we should run it on these libraries we should try and run it on all files and not exclude directories as that defeats the purpose of the checks.  If for some reason we do end up needing to exclude some files under the path then we should add the glob pattern as a template parameter.

Reply:
We are planning on unifying the pipelines in this repo to avoid this bifurcation.  @scbedd is aware of the work that needs to happen here.

Reply:
@weshaggard They are having two yml files which mutual exclude with each other.
Codes under sdk folder runs architype-sdk-client.yml
Codes outside of sdk folder (including the README.md in root) runs azure-pipeline.yml
The vendor folder is generated one during build time, so we cannot check the .md files under vendor. 

Do we want to check sdk folder under azure-pipeline.yml as well? I saw most of the steps exclude sdk path.

Reply:
No I'd suggest lets limit our link verification to archetype-sdk-client.yml for now and once things converge they will be get checked.

As for md files getting generated as part of the build those should still be checked and valid, especially if we plan to publish/release them anywhere. If the generate md files have bad links we should follow-up with the generation tools and get them fixed. We can of course do that as a follow-up item.

Reply:
Vendor includes 3rd party dependencies. We do not publish anything under vendor, so we cannot fix the broken links introduce by 3rd party. 

I'd suggest to leave the check in azure-pipeline as well until they find a way to unify the path. Otherwise, we have plenty of codes without coverage. @weshaggard thoughts?

Reply:
I'd be OK with leaving the check here but I'd still like to use our template, even if we need to make some tweaks to the template. Can we do the validation before the build so that we don't have pulled down all the 3rd party dependencies yet?

Reply:
I have no clues what's happened on the vm. But command failed to read the script under System root ({Build.SourceDirectory}/{system.default}), throw me an NoSuchFile exception.

Reply:
Tried before install dependencies and after. Not working anywhere.

Reply:
Ok I'm fine with keeping this calling the script directly for now, especially given that it will likely go away in the future.

Reply:
Out of curiosity does this work if you use a full path to the template like `/eng/common/pipelines/templates/steps/verify-links.yml`.

Reply:
`/eng/common/pipelines/templates/steps/verify-links.yml` works.
I have trouble to read `verify-links.ps1`. 
I made some experiment in test pipeline. It has trouble to read the `sdkPath` I passed to verify-links.yml.
https://dev.azure.com/azure-sdk/playground/_apps/hub/ms.vss-build-web.ci-designer-hub?pipelineId=1886&branch=azure-pipelines&nonce=R8Pr%2FoPlyhiu88lmCpfuzg%3D%3D

Reply:
Meanwhile, we have to change the template (Verify-Links.yml) to accommodate the changes.

Reply:
Code changes can refer here: https://github.com/sima-zhu/azure-sdk-for-go/tree/azure-pipelines

Reply:
Failed pipeline: https://dev.azure.com/azure-sdk/playground/_build/results?buildId=493797&view=logs&j=c30e86aa-a579-54ac-9cc0-7664eb300cda&t=e9bff205-e0a2-5082-2954-a81ea98353a2

Reply:
Passing:
```
      parameters:
        Workspace: '$(sdkPath)'
```
will not set `$(WorkSpace)` variable. You would need to reference it like `${{ parameters.WorkSpace }}`.

Reply:
By the way I'm fine with doing this as a follow-up I do think we might want to enable some more extension points to the template but we don't need to block on that fir this work.